src/query/interpreter.cpp

```diff
@@ -1871,6 +1879,40 @@ PreparedQuery PrepareIsolationLevelQuery(ParsedQuery parsed_query, bool in_expli
       RWType::NONE};
 }
 
+PreparedQuery PrepareStorageModeQuery(ParsedQuery parsed_query, const bool in_explicit_transaction,
+                                      InterpreterContext *interpreter_context) {
+  if (in_explicit_transaction) {
+    throw StorageModeModificationInMulticommandTxException();
+  }
+
+  auto *storage_mode_query = utils::Downcast<StorageModeQuery>(parsed_query.query);
+  MG_ASSERT(storage_mode_query);
+
+  const auto storage_mode = ToStorageStorageMode(storage_mode_query->storage_mode_);
+
+  auto callback = [storage_mode, interpreter_context]() -> std::function<void()> {
+    auto exists_active_transaction = interpreter_context->interpreters.WithLock([](const auto &interpreters_) {
+      return std::any_of(interpreters_.begin(), interpreters_.end(), [](const auto &interpreter) {
+        return interpreter->transaction_status_.load() == TransactionStatus::ACTIVE;
```

Comment:
interpreter_context contains all interpreters, including the one executing storage mode. If I am not wrong, you should include a check for yourself because this interpreter will certainly be active.

Reply:
I don't know if you remember, but we had a discussion about this during your development of the transactional queue. Since SetStorageModeQuery is not a Cypher query or Transaction query, the interpreter won't be active. So it won't be active.

Reply:
Yes, you are right I forgot about that :+1: